Hazel/Dtls/IRecordProtection.cs

```diff
@@ -29,7 +29,7 @@ public interface IRecordProtection : IDisposable
         /// <param name="output">Output ciphertext</param>
         /// <param name="input">Input plaintext</param>
         /// <param name="record">Parent DTLS record</param>
-        void EncryptServerPlaintext(ByteSpan output, ByteSpan input, ref Record record);
+        void EncryptServerPlaintext(ByteSpan output, ByteSpan input, ref Record record, ProtocolVersion protocolVersion);
```

Comment:
The protocol version should become part of the Record so we aren't passing it into these interface methods.

Reply:
Cool, I agree, that makes much more sense - changed!